coalib/parsing/Globbing.py

```diff
@@ -252,40 +252,47 @@ def _absolute_flat_glob(pattern):
     return
 
 
-def _iter_relative_dirs(dirname):
+def _iter_relative_dirs(dirname, ignored_globs=None):
     """
     Recursively iterates subdirectories of all levels from dirname
 
-    :param dirname: Directory name
-    :return:        Iterator that yields files and directory from the given dir
-                    and all it's (recursive) subdirectories
+    :param dirname:       Directory name
+    :param ignored_globs: List of globs to ignore when matching files
+    :return:              Iterator that yields files and directory from the
+                          given dir all it's (recursive) subdirectories
     """
     if not dirname:
         dirname = os.curdir
+
+    if ignored_globs:
+        if fnmatch(dirname, ignored_globs):
+            return
```

Comment:
@Makman2 looks like it is pretty hard to do that because of cache, so i would have to change lots of tests to make this work the way you asked. That final filter would better stay there